googleapi/googleapi.go

```diff
@@ -181,6 +180,22 @@ func CheckResponseWithBody(res *http.Response, body []byte) error {
 	}
 }
 
+// errorReplyFromBody attempts to get the error from body. The body
+// may be a JSON object or JSON array, or may be something else.
+func errorReplyFromBody(body []byte) (*errorReply, error) {
+	jerr := new(errorReply)
+	peekChar := body[0]
+	if peekChar == '[' {
```

Comment:
nit: Since we don't need the variable anywhere else, maybe just the following:
```suggestion
	if body[0] == '[' {
```

or `if peekChar := body[0]; peekChar == '[' {`

Reply:
Good catch, copy/paste error. Fixed.